File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/ModifyCalendarInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.utils.ImplementationHelper

data class ModifyCalendarInput(
    private val call: PluginCall,
) {
    val id: Long =
        call.getString("id")?.let { idString ->
            idString.toLongOrNull() ?: throw PluginError.InvalidCalendarId
        } ?: throw PluginError.CalendarIdMissing
    val title = call.getString("title")
    val color = call.getString("color")?.let { ImplementationHelper.hexToColorInt(it) }

    init {
        if (title == null && color == null) {
            throw PluginError.NoFieldsToModify
        }
    }
}

File: ios/Plugin/Models/Inputs/ModifyCalendarInput.swift
import Capacitor

struct ModifyCalendarInput {
    private let id: String
    private let title: String?
    private let color: CGColor?

    init(call: CAPPluginCall) throws {
        guard let id = call.getString("id") else {
            throw PluginError.calendarIdMissing
        }
        self.id = id
        self.title = call.getString("title")
        if let color = call.getString("color") {
            self.color = try UIColor.fromHex(color).cgColor
        } else {
            self.color = nil
        }
        if self.title == nil && self.color == nil {
            throw PluginError.noFieldsToModify
        }
    }

    func getId() -> String {
        return id
    }

    func getTitle() -> String? {
        return title
    }

    func getColor() -> CGColor? {
        return color
    }
}
